install/jstl/bin/ts.jtx

```diff
@@ -16,4 +16,46 @@
 
 #
 # $Id$
-#
+# https://github.com/jakartaee/tags/issues/255
```

Comment:
https://github.com/jakartaee/tags/issues/255#issuecomment-2025211892 discusses a possible code change to address the issue for EE 10+.  I'm not yet sure if the suggested code change to change the test based on which Java version is used, will help but it sounds like a worthwhile idea to try.

I'm thinking we should just leave the tags ts.jtx file empty for EE 11 since there are no Tags Spec updates for EE 11.

Reply:
> Updated: [jakartaee/tags#255 (comment)](https://github.com/jakartaee/tags/issues/255#issuecomment-2025211892) discusses a possible code change to address the issue for EE 10+. I'm not yet sure if the suggested code change to change the test based on which Java version is used, will help but it sounds like a worthwhile idea to try.
> 
> The only issue that I see is that the Platform TCK team doesn't have enough time to do everything that we are being asked to do. We should respond on the [jakartaee/tags#255](https://github.com/jakartaee/tags/issues/255) issue that we do not have time to make a test change but see if someone else wants to. Otherwise we will merge this change and the other EE 10 changes that are needed for Java 21 support.
> 
> Forget what I said about EE 11 as this pr is for the 10.0.x branch. :-)

Thanks Scott for the comments, As you rightly said the changes are applicable for complete release of Specification. The PR is for service release, where test changes is not possible. Please let me know if further changes are need in the PR.

Reply:
Quoting from https://jakarta.ee/committees/specification/tckprocess:

> A consensus that a test produces invalid results will result in the exclusion of that test from certification requirements, and an immediate update and release of an official distribution of the TCK including the new exclude list. The associated challenge issue MUST be closed with an accepted label to indicate it has been resolved.
> 
> The specification project may approve (user) workarounds for an accepted TCK challenge (as an alternative to excluding TCK tests).
> 
> As another alternative to excluding a challenged test, it may be possible to adjust the test validation logic to “expand” the validation check. E.g. if a test expects a value “A1” for a specific variable “x”, but a challenge is raised arguing that the specification language actually allows for either values “A1” and “A2” (but no other values) to be valid values for “x”, then it could be a valid course of action to modify the challenged test to allow either “A1” OR “A2” for “x”.
> 
> Since this line of thinking might be applied to cases that aren’t quite as straightforward as this example, care should be taken when using this approach. A particular danger is that an implementation that has already demonstrated compliance before the challenge was raised might actually not pass the new, modified test.
> 
> To limit the confusion and additional work such a scenario would cause, if there is already at least one certified compatible implementation before the challenge, the new, modified TCK should be run against at least one such implementation (and ideally all of them) before the changes are published, released, and finalized.
> 
> If such a change were released, and it was later found to cause a previously-certified implementation to fail the new, modified test, then excluding the test would likely be the only option, and this would require yet another, additional service release.
> 

More specifically, the text ` it may be possible to adjust the test validation logic to “expand” the validation check` mentions a specific use case that doesn't involve new releases of Java breaking compatibility but that is what broke the jstl tests.  To "expand" the (TCK) validation check we have to support both Java 11/17 (as before) but also "expand" the check to also work on Java 21.  To do that, I think there will be some separate files for the different Java versions and some test changes to use different expressions in the jstl tests.

Reply:
@gurunrao feel free to respond on the related EE 10 TCK challenge issue https://github.com/jakartaee/tags/issues/255#issuecomment-2100789831